compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -481,14 +480,6 @@ public void visit(BLangRecordTypeNode recordTypeNode) {
 
     @Override
     public void visit(BLangErrorType errorType) {
-        BType reasonType = getReasonType(errorType);
-
-        if (!types.isAssignable(reasonType, symTable.stringType)) {
-            dlog.error(errorType.reasonType.pos, DiagnosticCode.INVALID_ERROR_REASON_TYPE, reasonType);
-        } else if (errorType.reasonType != null) {
-            validateModuleQualifiedReasons(errorType.reasonType.pos, reasonType);
-        }
-
         if (errorType.detailType == null) {
```

Comment:
This is no longer valid, isn't it?

Reply:
We can still have something like `type F error`
https://ballerina.io/ballerina-spec/spec.html#section_5.5.1